packages/frontend/src/state/crab/hooks.ts

```diff
@@ -128,28 +130,43 @@ export const useCurrentCrabPositionValue = () => {
   const [currentCrabPositionValue, setCurrentCrabPositionValue] = useAtom(currentCrabPositionValueAtom)
   const [currentCrabPositionValueInETH, setCurrentCrabPositionValueInETH] = useAtom(currentCrabPositionValueInETHAtom)
   const { value: userCrabBalance } = useTokenBalance(crabStrategy, 15, 18)
-  const getWSqueethPositionValue = useGetWSqueethPositionValue()
-  const getWSqueethPositionValueInETH = useGetWSqueethPositionValueInETH()
   const contract = useAtomValue(crabStrategyContractAtom)
   const setCurrentEthLoading = useUpdateAtom(currentEthLoadingAtom)
+  const vault = useAtomValue(crabStrategyVaultAtom)
+  const getBuyQuote = useGetBuyQuote()
+  const ethPrice = useETHPrice()
+  const setStrategyData = useSetStrategyData()
+
+  useEffect(() => {
+    setStrategyData()
+  }, [])
 
   useAppEffect(() => {
-    ; (async () => {
+    ;(async () => {
       setIsCrabPositionValueLoading(true)
-      const squeethDebt = await getWsqueethFromCrabAmount(userCrabBalance, contract)
-      if (!squeethDebt) {
+      const [collateral, squeethDebt] = await Promise.all([
+        getCollateralFromCrabAmount(userCrabBalance, contract, vault),
+        getWsqueethFromCrabAmount(userCrabBalance, contract),
+      ])
+
+      if (!squeethDebt || !collateral) {
         setCurrentCrabPositionValue(BIG_ZERO)
         setCurrentCrabPositionValueInETH(BIG_ZERO)
         return
       }
-      setCurrentEthLoading(false)
-      const crabPositionValueInUSD = getWSqueethPositionValue(squeethDebt)
-      const crabPositionValueInETH = getWSqueethPositionValueInETH(squeethDebt)
+
+      const { amountIn: ethDebt } = await getBuyQuote(squeethDebt)
```

Comment:
Should not use quote, should get mid-price instead. There's a function in squeethPool hooks to convert sqth to eth

Reply:
@KMKoushik I just updated to use the the func in squeethPool